python/GafferSceneUI/OpenGLShaderUI.py

```diff
@@ -99,11 +100,11 @@ def shaderSubMenu() :
 	# a lot of irrelevancies at IE at the moment.
 	paths = [ os.environ["GAFFER_ROOT"] + "/glsl" ]
 	for path in paths :
-		for root, dirs, files in os.walk( path ) :
-			for file in files :
-				if os.path.splitext( file )[1] in ( ".vert", ".frag" ) :
-					shaderPath = os.path.join( root, file ).partition( path )[-1].lstrip( "/" )
-					shaders.add( os.path.splitext( shaderPath )[0] )
+		for extension in [ ".vert", ".frag" ] :
+			shaderPaths = pathlib.Path( path ).glob( "**/*" + extension )
+
+			for shaderPath in shaderPaths :
+				shaders.add( shaderPath.stem )
```

Comment:
Sorry, my original comment on this seems to have gone astray - maybe I forgot to push the button? I think we need to use `shaderPath.relative_to( path )` here, otherwise the recursive glob doesn't make sense (because `stem` throws away any intermediate directories).

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5538/commits/5b618002e951323d16a65537a8f602f7077ac5ed